packages/node-http-handler/src/node-http2-handler.ts

```diff
@@ -164,6 +168,9 @@ export class NodeHttp2Handler implements HttpHandler {
     if (existingSessions.length > 0 && !disableConcurrentStreams) return existingSessions[0];
 
     const newSession = connect(authority);
+    // AWS SDK does not expect server push streams, don't keep node alive without a request.
+    newSession.unref();
```

Comment:
Good catch! I don't find anywhere in the Node.js doc that a session is ref'd before making a request. Did you find it from your manual testing?

Reply:
Node docs are pretty inconsistent about what gets ref'd, but node is pretty consistent that only refs keep the process alive. Since the process didn't exit after making an http2 request, it was just a matter of checking the API and seeing that sessions have refs and giving it a try.